File: CORREDOR/Tipos Constructores/C#/MiProyecto/Novela.cs
public class Novela : Libro, AccionesLibro{
    public int Paginas { get; set; }
    public string Genero { get; set; }

    // Constructor con parámetros
    public Novela(string titulo, string autor, int paginas, string genero)
        : base(titulo, autor) // llama al constructor de Libro
    {
        Paginas = paginas;
        Genero = genero;
    }
    public void Abrir(){
        Console.WriteLine("Abriendo la novela...");
    }
    public override void Informacion(){
        Console.WriteLine($"Título: {Titulo}");
        Console.WriteLine($"Autor: {Autor}");
        Console.WriteLine($"Páginas: {Paginas}");
        Console.WriteLine($"Género: {Genero}");
    }
}

File: CORREDOR/Tipos Constructores/Go/Novela.go
package main

import "fmt"

type Novela struct {
	Libro
	Paginas int
	Genero  string
}

func NewNovela(titulo, autor string, paginas int, genero string) Novela {
	return Novela{
		Libro:   NewLibro(titulo, autor),
		Paginas: paginas,
		Genero:  genero,
	}
}

func (n Novela) Abrir() {
	fmt.Println("Abriendo la novela...")
}

func (n Novela) Informacion() {
	fmt.Println("Título:", n.Titulo)
	fmt.Println("Autor:", n.Autor)
	fmt.Println("Páginas:", n.Paginas)
	fmt.Println("Género:", n.Genero)
}